### ISSUE TITLE
Clients with `mysql_native_password` cannot connect to TiDB

### ISSUE THREAD

**Issue by djshow832** [pingcap/tidb#33141]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```go
dbaddr := fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?charset=utf8", "root", "", "127.0.0.1", 4000, "test")
mdb, err := sql.Open("mysql", dbaddr)
if err != nil {
    println("open error:", err.Error())
    return
}
err = mdb.Ping()
if err != nil {
    println("ping error:", err.Error())
    return
}
mdb.Close()
```

with 
```
require github.com/go-sql-driver/mysql v1.3.0
```

Note: mysql v1.6.0 is OK.

### 2. What did you expect to see? (Required)

No error

### 3. What did you see instead (Required)

ping error: this user requires mysql native password authentication.

### 4. What is your TiDB version? (Required)

master

This is caused by https://github.com/pingcap/tidb/pull/32338/files#diff-412615e1c1d4c6ae11a7885a11c879778bf401e0967c9c13c4b9dfd1f13ed29aR900

```go
if resp.Capability&mysql.ClientPluginAuth > 0 {              // Previously, there is another condition before it: resp.AuthPlugin != mysql.AuthNativePassword
	resp.AuthPlugin = mysql.AuthNativePassword
	authData, err := cc.authSwitchRequest(ctx, mysql.AuthNativePassword)
	if err != nil {
		return nil, err
	}
	return authData, nil
}
```

**Comment by djshow832** [comment]
I prefer to revert it for now because it's a compatibility breaker and release blocker.

**Comment by bb7133** [comment]
Thanks for reporting this. It's hard to say that this a bug of TiDB since I found that it is a known bug of `go-sql-driver` which has been fixed since `1.4.0`:

https://github.com/go-sql-driver/mysql/issues/785

However, I agree with @djshow832 with #33142 because:

* We do not know how many TiDB users are using `go-sql-driver 1.3.0` and this issue will break their workloads.
* The time for code freeze of v6.0 is close and it's hard to dig deeper for a 'fine solution'.

So we can revert the change temporarily and make a careful decision for it after v6.0.
